tests/cpp/test_sharding.cpp

```diff
@@ -462,6 +463,38 @@ TEST_F(ShardingTest, PropagationDoesNotOverwrite) {
   EXPECT_EQ(getShardedLoopAxis(tv3, ParallelType::DIDx), 0);
 }
 
+TEST_F(ShardingTest, BackpropagateToShardedTvs) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  DeviceMesh mesh = DeviceMesh::createForShape({2, 3});
+
+  TensorView* tv0 = makeContigTensor(2);
+  TensorView* tv1 = makeContigTensor(2);
+  TensorView* tv2 = set(tv1);
+  TensorView* tv3 = add(tv0, tv2);
+
+  tv0->setDeviceMesh(mesh);
+  tv0->outer_split(0, mesh.size(ParallelType::DIDx));
+  tv0->axis(0)->parallelize(ParallelType::DIDx);
+
+  tv1->setDeviceMesh(mesh);
+  tv1->outer_split(1, mesh.size(ParallelType::DIDy));
+  tv1->axis(1)->parallelize(ParallelType::DIDy);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  fusion->addOutput(tv3);
+
+  preseg_passes::OptimizationPass<
+      preseg_passes::PropagateShardingsPass>::runPass(fusion.get());
```

Comment:
Is it possible to run PreSegmenter instead, so the tests are more realistic? Other preseg passes could affect the end result. 

(Ideally, we should make them execution tests via FusionExecutorCache, but I guess you are trying to avoid having to fix downstream execution issues so this is totally fine.)